Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -3162,6 +3162,12 @@ GraphicsContext* RenderLayer::setupFilters(GraphicsContext& destinationContext,
 
 void RenderLayer::applyFilters(GraphicsContext& originalContext, const LayerPaintingInfo& paintingInfo, OptionSet<PaintBehavior> behavior, const LayerFragments& layerFragments)
 {
+    // Clipping is already applied if GrpahicsStyles are used.
```

Comment:
"GrpahicsStyles"

Reply:
I think this would be nicer as:
```
bool needsClipping = m_filters->hasSourceImage();
if (needsClipping) { .... };
m_filters->applyFilterEffect(originalContext);
```